org.adempiere.base/src/org/compiere/model/MCostDetail.java

```diff
@@ -864,6 +1148,37 @@ public boolean isDelta()
 			&& getDeltaQty().signum() == 0);
 	}	//	isDelta
 
+	
+	@Override
+	protected boolean beforeSave(boolean newRecord) {
+		if (newRecord) {
+			final String sql = "SELECT MAX(DateAcct) FROM M_CostDetail WHERE M_Product_ID=? AND Processed='Y'";
+			Timestamp MaxDateAcct = DB.getSQLValueTS(get_TrxName(), sql, getM_Product_ID());
+			Timestamp today = TimeUtil.getDay(System.currentTimeMillis());
+			if (MaxDateAcct != null && MaxDateAcct.after(today))
+				today = MaxDateAcct;
+			Timestamp dateAcct = getDateAcct();
+			if (dateAcct == null)
+			{
+				setDateAcct(today);
+				setIsBackDate(false);
+			}
+			else
+			{
+				setDateAcct(dateAcct);
+				setIsBackDate(dateAcct.before(today));
+			}
+		} else {
+			if (is_ValueChanged(COLUMNNAME_DateAcct))
+			{
+				log.saveError("Error", Msg.getMsg(getCtx(), "CannotChangeProcessedDate"));
```

Comment:
misleading error message here

Reply:
Created a new error message: CannotChangeAccountDate - Account Date cannot be changed